cmd/kubelet/app/server.go

```diff
@@ -592,50 +592,57 @@ func run(ctx context.Context, s *options.KubeletServer, kubeDeps *kubelet.Depend
 	}
 
 	// if in standalone mode, indicate as much by setting all clients to nil
-	switch {
-	case standaloneMode:
+	if standaloneMode {
 		kubeDeps.KubeClient = nil
 		kubeDeps.EventClient = nil
 		kubeDeps.HeartbeatClient = nil
 		klog.InfoS("Standalone mode, no API client")
-
-	case kubeDeps.KubeClient == nil, kubeDeps.EventClient == nil, kubeDeps.HeartbeatClient == nil:
+	} else if kubeDeps.KubeClient == nil ||
+		kubeDeps.EventClient == nil ||
+		kubeDeps.HeartbeatClient == nil {
 		clientConfig, closeAllConns, err := buildKubeletClientConfig(ctx, s, nodeName)
 		if err != nil {
 			return err
 		}
 		if closeAllConns == nil {
 			return errors.New("closeAllConns must be a valid function other than nil")
 		}
-		kubeDeps.OnHeartbeatFailure = closeAllConns
 
-		kubeDeps.KubeClient, err = clientset.NewForConfig(clientConfig)
-		if err != nil {
-			return fmt.Errorf("failed to initialize kubelet client: %w", err)
-		}
+		if kubeDeps.KubeClient == nil {
+			kubeDeps.OnHeartbeatFailure = closeAllConns
```

Comment:
why is this inside the `if kubeDeps.KubeClient == nil` block?

Reply:
/hold

this looks like a potential bug

Reply:
@liggitt is right, this override as currently implemented is general when either `kubeDeps.KubeClient`, `kubeDeps.EventClient`, _or_ `kubeDeps.HeartbeatClient` is nil, it shouldn't be overriden *only* in the `kubeDeps.KubeClient == nil` scenario.

Fixed.

Thanks for catching this!

Reply:
@klueska @mrunalp could you take another look at this (and thanks for your stamina!)